containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -36,6 +37,29 @@
 MAX_IDENTIFIER_LENGTH = 255
 
 
+class MetadataTypes(Enum):
+    """
+    Class to identify different types of metadata.
+
+    data product metadata - relates to data of the data product as a whole. Can be 1 or more tables.
+
+    and
+
+    data product schema - relates to data describing a table within a data product
+    """
+    data_product_metadata: str = "metadata"
+    data_product_schema: str = "schema"
+
+    @classmethod
+    def _missing_(cls, value):
```

Comment:
Is this `__missing__` just to customise the error message? I get a ValueError by default if I call it with an undefined value, just a slightly less helpful one

```python
>>> MetadataTypes('xyz')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/matt.moore/.pyenv/versions/3.11.5/lib/python3.11/enum.py", line 712, in __call__
    return cls.__new__(cls, value)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matt.moore/.pyenv/versions/3.11.5/lib/python3.11/enum.py", line 1128, in __new__
    raise ve_exc
ValueError: 'xyz' is not a valid MetadataTypes
```

Reply:
yeah it is